docs/sources/developers/http_api/datasource_permissions.md

```diff
@@ -268,31 +271,62 @@ Status codes:
 - **400** - Permission cannot be added, see response body for details
 - **401** - Unauthorized
 - **403** - Access denied
-- **404** - Datasource not found
 
-## Remove permission for a data source
+## Add or revoke access to a data source for a basic role
+
+`POST /api/access-control/datasources/:uid/builtInRoles/:builtinRoleName`
 
-`DELETE /api/datasources/:id/permissions/:permissionId`
+Sets permission for the data source with the given `uid` to all users who have the specified basic role.
 
-Removes the permission with the given `permissionId` for the data source with the given `id`.
+You can set permissions for the following basic roles: `Admin`, `Editor`, `Viewer`.
+
+To add a permission, set the `permission` field to either `Query` or `Edit`.
+To remove a permission, set the `permission` field to an empty string.
 
 **Required permissions**
 
 See note in the [introduction]({{< ref "#data-source-permissions-api" >}}) for an explanation.
 
-| Action                        | Scope                                                                        |
-| ----------------------------- | ---------------------------------------------------------------------------- |
-| datasources.permissions:write | datasources:\*<br>datasources:id:\*<br>datasources:id:1 (single data source) |
+| Action                        | Scope                                                                                      |
+| ----------------------------- | ------------------------------------------------------------------------------------------ |
+| datasources.permissions:write | datasources:\*<br>datasources:uid:\*<br>datasources:uid:my_datasource (single data source) |
 
 ### Examples
 
 **Example request:**
 
 ```http
-DELETE /api/datasources/1/permissions/2
+POST /api/access-control/datasources/my_datasource/builtInRoles/Admin
+Accept: application/json
+Content-Type: application/json
+Authorization: Bearer eyJrIjoiT0tTcG1pUlY2RnVKZTFVaDFsNFZXdE9ZWmNrMkZYbk
+
+{
+  "permission": "Edit",
+}
+```
+
+**Example response:**
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json; charset=UTF-8
+Content-Length: 35
+
+{"message": "Permission updated"}
+```
+
+**Example request:**
+
+```http
+POST /api/access-control/datasources/my_datasource/builtInRoles/Viewer
 Accept: application/json
 Content-Type: application/json
 Authorization: Bearer eyJrIjoiT0tTcG1pUlY2RnVKZTFVaDFsNFZXdE9ZWmNrMkZYbk
+
+{
+  "permission": "",
```

Comment:
should it be some response herre?

Reply:
This is the case for removing a permission - to remove it we pass in
```
{"permission": ""}
```
and get back
```
{"message": "Permission removed"}
```